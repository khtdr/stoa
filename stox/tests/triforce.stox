#!/usr/bin/env stox

fun triforce() {
  print "triforce:
 ▲
▲ ▲";
  return;
}

triforce();

var i = 0;
var j = 0;

while (true)
  while (i < 7) {
    j = i;
    i = i + 1;
    while (j < 7) {
      j = j + 1;
      if (j == 1) continue;
      if (j == 2) break;
      print "i,j " + i + "," + j;
      if (j == 4) continue 2;
      print "-";
      if (j == 5) break;
      if (j == 6) break 3;
    }
    print "X";
  }

print "done!";
