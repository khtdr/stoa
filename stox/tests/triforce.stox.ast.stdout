(program 
   (fun triforce (let [] (block 
      (print "triforce:\n ▲\n▲ ▲")
      (return nil)
   )))
   (call triforce)
   (var i 0)
   (var j 0)
   (while true 
      (while (< i 7) 
         (block 
            (= j i)
            (= i (+ i 1))
            (while (< j 7) 
               (block 
                  (= j (+ j 1))
                  (if (== j 1) (CONTINUE 1))
                  (if (== j 2) (BREAK 1))
                  (print (+ (+ (+ "i,j " i) ",") j))
                  (if (== j 4) (CONTINUE 2))
                  (print "-")
                  (if (== j 5) (BREAK 1))
                  (if (== j 6) (BREAK 3))
               )
            )
            (print "X")
         )
      )
   )
   (print "done!")
)
